src/dfuTrigger.js

```diff
@@ -165,13 +165,27 @@ const getDfuInfo = openDecorator((usbdev, interfaceNumber) => (
 ));
 
 const sendDetachRequest = openDecorator((usbdev, interfaceNumber) => (
-    new Promise(resolve => {
+    new Promise((resolve, reject) => {
         assertDfuTriggerInterface(usbdev, interfaceNumber);
         usbdev.controlTransfer(
             ReqTypeOUT, DFU_DETACH_REQUEST, 0, interfaceNumber, detachReqBuf,
-            () => {
-                // Device is expected to disappear, any further calls would fail
-                resolve();
+            (err, data) => {
+                // If the detach is successful, the target device will reboot
+                // before sending a response, so the expected result is that
+                // the control transfer will stall.
+                if (err &&
+                    err.errno === usb.LIBUSB_TRANSFER_STALL &&
```

Comment:
Why is it necessary to check both err.errno and err.message?

Reply:
It's not necessary, maybe I'm being overcautious here. `errno` should be either a number from [`enum libusb_transfer_status`](http://libusb.sourceforge.net/api-1.0/group__asyncio.html#ga9fcb2aa23d342060ebda1d0cf7478856) or a negative number from [enum libusb_error](http://libusb.sourceforge.net/api-1.0/group__libusb__misc.html#gab2323aa0f04bc22038e7e1740b2f29ef) (in the JS bindings, anything non-zero gets converted into an error).

It *should* be safe to have only one check, if you want.